internal/clusterfeature/clusterfeatureadapter/gorm_feature_repository.go

```diff
@@ -77,24 +77,23 @@ func (cfm clusterFeatureModel) String() string {
 	return fmt.Sprintf("Id: %d, Creation date: %s, Name: %s", cfm.ID, cfm.CreatedAt, cfm.Name)
 }
 
-// gormFeatureRepository component in charge for executing persistence operation on Features.
+// GORMFeatureRepository implements feature persistence in RDBMS using GORM.
 // TODO: write integration tests
-type gormFeatureRepository struct {
-	db *gorm.DB
-
+type GORMFeatureRepository struct {
+	db     *gorm.DB
 	logger common.Logger
 }
 
-// NewGormFeatureRepository returns a feature repository persisting feature state into database using Gorm.
-func NewGormFeatureRepository(db *gorm.DB, logger common.Logger) clusterfeature.FeatureRepository {
-	return &gormFeatureRepository{
-		db: db,
-
+// NewGormFeatureRepository returns a feature repository persisting feature state into database using GORM.
+func NewGormFeatureRepository(db *gorm.DB, logger common.Logger) GORMFeatureRepository {
```

Comment:
```suggestion
func NewGormFeatureRepository(db *gorm.DB, logger common.Logger) *GORMFeatureRepository {
```

Reply:
I think that this is unnecessary. As a performance optimization it would be premature. As for instance sharing, you could make that decision on the caller's side by taking the address of the returned instance and passing that around. Also, instance sharing is only relevant if you mutate the instance, which this implementation does not do. See also: https://stackoverflow.com/a/23551970/578530